File: 2279.装满石头的背包的最大数量.go
/*
 * @lc app=leetcode.cn id=2279 lang=golang
 *
 * [2279] 装满石头的背包的最大数量
 */

// @lc code=start
func maximumBags(capacity []int, rocks []int, additionalRocks int) int {
	for i := range capacity {
		capacity[i] -= rocks[i]
	}
	sort.Ints(capacity)
	res := 0
	for i := range capacity {
		if additionalRocks >= capacity[i] {
			additionalRocks -= capacity[i]
			res++
		} else {
			break
		}
	}
	return res
}

// @lc code=end

File: 2279.装满石头的背包的最大数量.py
#
# @lc app=leetcode.cn id=2279 lang=python3
#
# [2279] 装满石头的背包的最大数量
#
from typing import List
# @lc code=start
class Solution:
    def maximumBags(self, capacity: List[int], rocks: List[int], additionalRocks: int) -> int:
        abs_res = [0] * len(capacity)
        for i in range(len(capacity)):
            abs_res[i] = capacity[i] - rocks[i]
        abs_res.sort()
        res = 0
        for i in range(len(abs_res)):
            if additionalRocks >= abs_res[i]:
                additionalRocks -= abs_res[i]
                res += 1
            else:
                break
        return res
# @lc code=end

